integration-test/spark-common/src/test/java/com/tencent/rss/test/RepartitionWithMemoryRssTest.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * Tencent is pleased to support the open source community by making
+ * Firestorm-Spark remote shuffle server available.
+ *
+ * Copyright (C) 2021 THL A29 Limited, a Tencent company.  All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use
+ * this file except in compliance with the License. You may obtain a copy of the
+ * License at
+ *
+ * https://opensource.org/licenses/Apache-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package com.tencent.rss.test;
+
+import com.google.common.io.Files;
+import com.tencent.rss.coordinator.CoordinatorConf;
+import com.tencent.rss.server.ShuffleServerConf;
+import com.tencent.rss.storage.util.StorageType;
+import org.apache.spark.SparkConf;
+import org.apache.spark.shuffle.RssClientConfig;
+import org.junit.BeforeClass;
+
+import java.io.File;
+
+public class RepartitionWithMemoryRssTest extends RepartitionTest {
+
+  @BeforeClass
+  public static void setupServers() throws Exception {
+    CoordinatorConf coordinatorConf = getCoordinatorConf();
+    coordinatorConf.set(CoordinatorConf.COORDINATOR_APP_EXPIRED, 5000L);
+    createCoordinatorServer(coordinatorConf);
+    ShuffleServerConf shuffleServerConf = getShuffleServerConf();
+    shuffleServerConf.set(ShuffleServerConf.SERVER_HEARTBEAT_INTERVAL, 5000L);
+    shuffleServerConf.set(ShuffleServerConf.SERVER_APP_EXPIRED_WITHOUT_HEARTBEAT, 4000L);
+    File tmpDir = Files.createTempDir();
+    tmpDir.deleteOnExit();
+    File dataDir1 = new File(tmpDir, "data1");
+    File dataDir2 = new File(tmpDir, "data2");
+    String basePath = dataDir1.getAbsolutePath() + "," + dataDir2.getAbsolutePath();
+    shuffleServerConf.set(ShuffleServerConf.RSS_STORAGE_TYPE, StorageType.LOCALFILE.name());
+    shuffleServerConf.set(ShuffleServerConf.RSS_STORAGE_BASE_PATH, basePath);
+    shuffleServerConf.set(ShuffleServerConf.SERVER_MEMORY_SHUFFLE_ENABLED, true);
+    shuffleServerConf.setString(ShuffleServerConf.SERVER_BUFFER_CAPACITY.key(), "512mb");
+    shuffleServerConf.setString(ShuffleServerConf.SERVER_MEMORY_SHUFFLE_LOWWATERMARK.key(), "100mb");
+    shuffleServerConf.setString(ShuffleServerConf.SERVER_MEMORY_SHUFFLE_HIGHWATERMARK.key(), "400mb");
+    createShuffleServer(shuffleServerConf);
+    startServers();
+  }
+
+  @Override
+  public void updateRssStorage(SparkConf sparkConf) {
+    sparkConf.set(RssClientConfig.RSS_STORAGE_TYPE, "MEMORY,LOCALFILE");
```

Comment:
MEMORY_AND_LOCAILFILE?

Reply:
Current style for flexible without adding new storage type, eg, "MEMORY,HDFS" means memory and hdfs, "LOCAL,HDFS" means local and hdfs

Reply:
I think multi storage should separate from single storage. MEMORY_AND_LOCALFILE is more Spark's style.And i can't find more discussion on this in Spark's pr.

Reply:
Agree to refactor multi storage, I think it can reuse ComposedClientReadHandler, but it should be another PR

Reply:
storageType need to be split with `,`. It's a little weird.